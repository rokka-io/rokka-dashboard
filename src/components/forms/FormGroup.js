import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'
import RequiredIndicator from './RequiredIndicator'

const FormGroup = ({ label, children, className = null, required = false, error = null }) => (
  <div className={cx('rka-form-group', className, {'has-error': !!error})}>
    <label className="rka-label">
      {label}
      <RequiredIndicator required={required} />
    </label>
    {children}
    {error ? <div className="mt-xs txt-xs txt-cranberry">{error}</div> : null}
  </div>
)
FormGroup.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  required: PropTypes.bool,
  error: PropTypes.string
}

export default FormGroup
