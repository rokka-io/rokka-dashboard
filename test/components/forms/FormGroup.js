import React from 'react'
import FormGroup from '../../../src/components/forms/FormGroup'
import renderer from 'react-test-renderer'

test('FormGroup does render', () => {
  const component = renderer.create(
    <FormGroup label="Test">
      <span>Test</span>
    </FormGroup>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('FormGroup does render with required indicator', () => {
  const component = renderer.create(
    <FormGroup label="Test" required>
      <span>Test</span>
    </FormGroup>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('FormGroup does render with className', () => {
  const component = renderer.create(
    <FormGroup label="Test" className="test">
      <span>Test</span>
    </FormGroup>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('FormGroup does render with error', () => {
  const component = renderer.create(
    <FormGroup label="Test" error="Test">
      <span>Test</span>
    </FormGroup>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
