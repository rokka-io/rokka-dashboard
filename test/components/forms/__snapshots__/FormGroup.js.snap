exports[`test FormGroup does render 1`] = `
<div
  className="rka-form-group">
  <label
    className="rka-label">
    Test
  </label>
  <span>
    Test
  </span>
</div>
`;

exports[`test FormGroup does render with className 1`] = `
<div
  className="rka-form-group test">
  <label
    className="rka-label">
    Test
  </label>
  <span>
    Test
  </span>
</div>
`;

exports[`test FormGroup does render with error 1`] = `
<div
  className="rka-form-group has-error">
  <label
    className="rka-label">
    Test
  </label>
  <span>
    Test
  </span>
  <div
    className="mt-xs txt-xs txt-cranberry">
    Test
  </div>
</div>
`;

exports[`test FormGroup does render with required indicator 1`] = `
<div
  className="rka-form-group">
  <label
    className="rka-label">
    Test
    <span>
       *
    </span>
  </label>
  <span>
    Test
  </span>
</div>
`;
